src/neo/SmartContract/ApplicationEngine.Storage.cs

```diff
@@ -15,9 +15,9 @@ partial class ApplicationEngine
         public static readonly InteropDescriptor System_Storage_AsReadOnly = Register("System.Storage.AsReadOnly", nameof(AsReadOnly), 1 << 4, CallFlags.ReadStates);
         public static readonly InteropDescriptor System_Storage_Get = Register("System.Storage.Get", nameof(Get), 1 << 15, CallFlags.ReadStates);
         public static readonly InteropDescriptor System_Storage_Find = Register("System.Storage.Find", nameof(Find), 1 << 15, CallFlags.ReadStates);
-        public static readonly InteropDescriptor System_Storage_Put = Register("System.Storage.Put", nameof(Put), 0, CallFlags.WriteStates);
-        public static readonly InteropDescriptor System_Storage_PutEx = Register("System.Storage.PutEx", nameof(PutEx), 0, CallFlags.WriteStates);
-        public static readonly InteropDescriptor System_Storage_Delete = Register("System.Storage.Delete", nameof(Delete), 0, CallFlags.WriteStates);
+        public static readonly InteropDescriptor System_Storage_Put = Register("System.Storage.Put", nameof(Put), 0, CallFlags.States);
+        public static readonly InteropDescriptor System_Storage_PutEx = Register("System.Storage.PutEx", nameof(PutEx), 0, CallFlags.States);
+        public static readonly InteropDescriptor System_Storage_Delete = Register("System.Storage.Delete", nameof(Delete), 0, CallFlags.States);
```

Comment:
I'd probably leave these as is.

Reply:
We need to read the `IsConstant` field before writing. Do we need `IsConstant`? Maybe we can remove it.

Reply:
@shargon @roman-khimov What do you think?

Reply:
It can be a significant storage item simplification, but we'll loose a safety feature at the same time that I can't see any substitute for (other than a separate "constant storage context", but that's not really better than the flag we have at the moment). But I'm not sure it's really used by contracts.

Reply:
I think it's never used.